---
title: "Data_Preprocessing"
output: html_document
date: '2023-03-14'
---

```{r setup, include=FALSE}
reqpkg = c("reshape2", "cluster", "doParallel", "foreach", 
           "ggplot2", "grid", "scales", "metagenomeSeq", "plyr",  "ROCR", "tidyr", "SpiecEasi",
           "breakaway", "DivNet", "vegan", "lubridate", "tidyverse", "DirichletReg", "phyloseq"
           #"metaSPARSim", "GMPR", "MASS"
           )
source("utils_preprocess.R")
# Load all required packages and show version
for(i in reqpkg){
	print(i); print(packageVersion(i))
	library(i, quietly=TRUE, verbose=FALSE, warn.conflicts=FALSE, character.only=TRUE)
}
```

# Full Dimensional Setup

## Load in Data

```{r data loading}
load("genus.Rdata")
load("phy.Rdata")
bodyweight=read.csv("FinalBodyWeight.csv")
#phy.genus <- phy.norm

# this I have changed on 21st Sept 2021
phy.genus <- phy.norm

phy.sub = phy.genus %>%
  subset_samples(Timepoint%in%c("Day_21"))

agg.level = "Genus"

phy.order = tax_glom(phy.sub, agg.level)
suffix = paste0(agg.level, "_230813")

phy.org <- phy.order

otu_gut <- otu_table(phy.order)@.Data
tax <- tax_table(phy.order)

identical(rownames(otu_gut),rownames(tax))

nam.tem=tax[,agg.level]
```


```{r}
phy.sub
```

```{r}
phy.order
```


```{r}
library(stringr)

if (agg.level=="Phylum") {
  tree_name=paste(str_sub(tax[, "Kindom"], start=4), 
                  str_sub(tax[, "Phylum"], start=4), sep="*")
}

if (agg.level=="Class") {
  tree_name=paste(str_sub(tax[, "Kindom"], start=4), 
                  str_sub(tax[, "Phylum"], start=4), 
                  str_sub(tax[, "Class"], start=4), sep="*")
}

if (agg.level=="Order") {
  tree_name=paste(str_sub(tax[, "Kindom"], start=4), 
                  str_sub(tax[, "Phylum"], start=4), 
                  str_sub(tax[, "Class"], start=4), 
                  str_sub(tax[, "Order"], start=4), sep="*")
}

if (agg.level=="Family") {
  tree_name=paste(str_sub(tax[, "Kindom"], start=4), 
                  str_sub(tax[, "Phylum"], start=4), 
                  str_sub(tax[, "Class"], start=4), 
                  str_sub(tax[, "Order"], start=4), 
                  str_sub(tax[, "Family"], start=4),sep="*")
}

if (agg.level=="Genus") {
  tree_name=paste(str_sub(tax[, "Kindom"], start=4), 
                  str_sub(tax[, "Phylum"], start=4), 
                  str_sub(tax[, "Class"], start=4), 
                  str_sub(tax[, "Order"], start=4), 
                  str_sub(tax[, "Family"], start=4),
                  str_sub(tax[, "Genus"], start=4), sep="*")
}


```

```{r}
rownames(otu_gut) = tree_name

sam <- sample_data(phy.order)
sam.2<-as.data.frame(slot(sam,".Data"))
colnames(sam.2)<-colnames(sam)
sam.incl.weight <- merge(sam.2, bodyweight, by.x="MouseID", by.y="Sample.ID")
sam.weight <- sam.incl.weight[c("X.SampleID", "weight")]
sam.weight <- sam.weight %>% remove_rownames %>% column_to_rownames(var="X.SampleID")

org.save <- t(otu_gut)
org.save <- merge(org.save, sam.weight, by=0)
org.save <- subset(org.save,select=-c(Row.names))
org.save <- t(org.save)

sam$name <- paste0(sam$Treatment, seq(1, dim(sam)[1]))
sam$sampleNum <- seq(1, dim(sam)[1])
colnames(org.save) <- sam$name
```
## Save Data for Analysis

```{r}
write.csv(org.save,
            paste0("/Users/elisabeth.ailer/Projects/P1_Microbiom/Code/input/data/orgData_Day21_",
                   suffix, ".csv"))

```


